docker-compose.yml

```diff
@@ -24,13 +24,16 @@ services:
     image: postgres:13
     restart: unless-stopped
     environment:
+      PGPORT: ${DB_PORT:-5432}
```

Comment:
This should address the concern @Gummibeer pointed out.

Reply:
Could you do this for redis as well ?

Reply:
Also did you faced any problems? I dont think its required ..

Reply:
Absolutely can do that.

Yes I did encounter problems without this change.
@gummibeer was right that inside the container the ports were still wrong.

Reply:
No my question is .. did you faced any problems while migrate seed or any other operation !! Its not a problem what we use internally..

Reply:
All set now!

Reply:
Sorry, I dont see need of these changes as of now ..!

Reply:
I've already mentioned that **without this** there are errors. These changes are necessary.
Or like @gummibeer suggested you need to add new fields to `.env` for `DOCKER_` specific vars.

Reply:
Since you didn't take my word for it....
![Screen Shot 2021-07-16 at 2 01 06 PM](https://user-images.githubusercontent.com/619938/125990033-761f3a47-0dd1-4f97-8c1d-c7557fb077b9.png)
![Screen Shot 2021-07-16 at 2 00 50 PM](https://user-images.githubusercontent.com/619938/125990035-e905c2ef-e50b-42ed-89ea-5f9dcfc25ba7.png)
![Screen Shot 2021-07-16 at 2 00 33 PM](https://user-images.githubusercontent.com/619938/125990036-8dc15fb2-0266-4414-8731-f5321c9741f7.png)

Reply:
As you see in the screen shots - by only changing the external port based on the Apps `DB_PORT` things break.
So either:

1. Change all the ports uniformly inside and outside the container,
2. Only change the exposed port based on a new `DOCKER_` variable in `.env`, or
3. Choose not to expose the service ports at all and only run artisan commands inside the container.

Reply:
I was able to reproduce it.  Its correct although, expose is not needed as you already is exposing port..

Reply:
AFAIK it is. `expose` is a way to expose to other containers.
Where as `ports` is used to expose the service to the host.

Reply:
To your knowledge - would using `ports` only instead of both also expose the port to all neighboring containers?

Reply:
can you explain what do you mean by this `"${DB_PORT:-5432}:${DB_PORT:-5432}"` and `PGPORT: ${DB_PORT:-5432}` ?

Reply:
1. The Apps "Laravel" .env file which is also used by docker:
![Screen Shot 2021-07-16 at 3 20 10 PM](https://user-images.githubusercontent.com/619938/125998353-b0d511bb-f1c5-4373-88ff-1d8b04556752.png)

2. We tell the postgress container to use the correct port matching what Laravel tries to use:
![Screen Shot 2021-07-16 at 3 22 13 PM](https://user-images.githubusercontent.com/619938/125998580-2673f593-ab17-4fda-8822-bb7edb677a48.png)

3. Because the internal postgres port now matches the port laravel config expects, the old mapping you used of `"${DB_PORT:-5432}:5432"` is now incorrect. The external exposed port for the container is correct, but it maps to an internal port with nothing running on it.

Reply:
Just check the above msg again..

Reply:
I have removed the `expose` lines as requested.

Reply:
> what do you mean by this "${DB_PORT:-5432}:${DB_PORT:-5432}" and PGPORT: ${DB_PORT:-5432} ?

I've read this a few times and feel that I answered the question. If you don't think I have please ask a more specific query. I simply do not understand the misunderstanding here.

`PGPORT` is an ENV variable used by the underlying image for the postgres container. We set that to either the value provided by the `.env` or the default value. We then mirror this change by exposing the containers internal port on the same port for the host machine.

Reply:
At this point I'm just going to "tap out" and close this PR.
You can address any of these issues however you prefer.